File: android/measure/src/main/java/sh/measure/android/attributes/SessionAttributeProcessor.kt
package sh.measure.android.attributes

import sh.measure.android.SessionManager

/**
 * Generates the session start time attribute. This attribute changes when a new session is created,
 * so it is computed every time [appendAttributes] is called.
 */
internal class SessionAttributeProcessor(
    private val sessionManager: SessionManager,
) : AttributeProcessor {
    override fun appendAttributes(attributes: MutableMap<String, Any?>) {
        attributes.put(Attribute.SESSION_START_TIME_KEY, sessionManager.getSessionStartTime())
    }
}

File: ios/Sources/MeasureSDK/Swift/Attribute/SessionAttributeProcessor.swift
//
//  SessionAttributeProcessor.swift
//  Measure
//
//  Created by Adwin Ross on 02/03/26.
//

import Foundation

final class SessionAttributeProcessor: AttributeProcessor {
    private let sessionManager: SessionManager
    private let timeProvider: TimeProvider
    
    init(sessionManager: SessionManager, timeProvider: TimeProvider) {
        self.sessionManager = sessionManager
        self.timeProvider = timeProvider
    }

    func appendAttributes(_ attribute: Attributes) {
        if let sessionStartTime = sessionManager.getSessionStartTime() {
            attribute.sessionStartTime = timeProvider.iso8601Timestamp(timeInMillis: sessionStartTime)
        }
    }
}
